File: nearly-signalr-server/Domain/UserRelationship.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data;
using Helper;

namespace Domain
{
    public class UserRelationship
    {

        [MaxLength(36)]
        [ForeignKey(nameof(User))]
        [InverseProperty(nameof(User))]
        public string RequesterId { get; set; } = default!;
        public User? Requester { get; set; }
        
        [MaxLength(36)]
        [ForeignKey(nameof(User))]
        [InverseProperty(nameof(User))]
        public string ResponderId { get; set; } = default!;
        public User? Responder { get; set; }
        
        public string RequesterName { get; set; } = default!;
        public string ResponderName { get; set; } = default!;
        public UserRelationshipStatus Status { get; set; } = default!;
    }
}

File: nearly-spring-api/src/main/java/com/ukrainianboyz/nearly/db/entity/UserRelationship.java
package com.ukrainianboyz.nearly.db.entity;


import com.ukrainianboyz.nearly.db.enums.Status;
import lombok.*;

import javax.persistence.*;

@AllArgsConstructor
@Entity
@Table(name = "\"UserRelationships\"", schema = "nearlyDB")
@NoArgsConstructor
@Data
@Builder
@IdClass(UserRelationshipPK.class)
public class UserRelationship {

    @Id
    @Column(name = "RequesterId")
    private String requesterId;

    @Id
    @Column(name = "ResponderId")
    private String responderId;

    @Column(name = "RequesterName")
    private String requesterName;

    @Column(name = "ResponderName")
    private String responderName;

    @Column(name = "Status")
    @Enumerated(EnumType.ORDINAL)
    private Status status;

    public String getOtherId(String id){
        return requesterId.equals(id) ? responderId : requesterId;
    }
}
